index.bs

```diff
@@ -535,101 +539,233 @@ dictionary WriteParams {
   (BufferSource or Blob or USVString)? data;
 };
 
+typedef (BufferSource or Blob or USVString or WriteParams) FileSystemWriteChunkType;
+
 [Exposed=(Window,Worker), SecureContext]
 interface FileSystemWritableFileStream : WritableStream {
-  Promise<void> write((BufferSource or Blob or USVString or WriteParams) data);
+  Promise<void> write(FileSystemWriteChunkType data);
   Promise<void> seek(unsigned long long position);
   Promise<void> truncate(unsigned long long size);
 };
 </xmp>
 
-A {{FileSystemWritableFileStream}} object is a {{WritableStream}} object with additional methods to operate
-on a single file on disk.
-
-Upon creation, an underlying sink will have been created and the stream will be usable. All operations executed on the stream are queuable and producers will be able to respond to backpressure.
+A {{FileSystemWritableFileStream}} has an associated <dfn for=FileSystemWritableFileStream>\[[file]]</dfn> (a [=file entry=]).
 
-The underlying sink's write method, and therefore {{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}} method, will accept byte-like data or {{WriteParams}} as input.
+A {{FileSystemWritableFileStream}} has an associated <dfn for=FileSystemWritableFileStream>\[[buffer]]</dfn> (a [=byte sequence=]).
+It is initially empty.
 
-The {{FileSystemWritableFileStream}} has a file position cursor initialized at byte offset 0 from the top of the file. When using {{FileSystemWritableFileStream/write()|write()}} or by using WritableStream capabilities through the {{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}} method, this position will be advanced based on the number of bytes written through the stream object.
+Note: This buffer can get arbitrarily large, so it is expected that implementations will not keep this in memory,
+but instead use a temporary file for this.
```

Comment:
Ah, this helps explain why createWritable() returns a promise. It might be good to note that all access to [[buffer]] is done in promise-returning methods, so even though it looks sync in the spec, implementations can make it async.

Reply:
The other reason createWritable() returns a promise is the "TODO: Check permissions.", in particular if the user doesn't already have write permission trying to create a writable will prompt for that permission.

Anyway, I agree that adding such a sentence about [[buffer]] is helpful, so I'll do that.